src/js_realm.hpp

```diff
@@ -1041,21 +1007,67 @@ void RealmClass<T>::object_for_object_id(ContextType ctx, ObjectType this_object
     if (!sync::has_object_ids(realm->read_group()))
         throw std::logic_error("Realm._objectForObjectId() can only be used with synced Realms.");
 
-    std::string object_type = Value::validated_to_string(ctx, args[0]);
-    validated_object_schema_for_value(ctx, realm, args[0], object_type);
-
+    auto& object_schema = validated_object_schema_for_value(ctx, realm, args[0]);
     std::string object_id_string = Value::validated_to_string(ctx, args[1]);
-    auto object_id = object_id_from_string(object_id_string);
+    auto object_id = sync::ObjectID::from_string(object_id_string);
```

Comment:
Can you access `sync::ObjectID` when `REALM_ENABLE_SYNC` isn't defined?

Reply:
This code is guarded by `REALM_ENABLE_SYNC`.

Reply:
Oh yeah, line 1005